### PR TITLE
Create profile preview

### PR DESCRIPTION
**Pull request by montakarlo**

### CODE REVIEW

File: frontend/src/components/ProfilePreview/index.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  user: IUserState;
+  trigger: () => React.ReactElement;
+  id: string;
+  onSend: (message: string) => void;
+}
+
+// const ProfilePreview: FunctionComponent<IProps> = ({ user, trigger, id, onSend }) => {
+const ProfilePreview: FunctionComponent = () => {
+  // Mocked data: testData and trigger function
+  const testData = {
```
Comment (vlvereta) [inline frontend/src/components/ProfilePreview/index.tsx:21]: Place mocked data outside of the component

File: frontend/src/components/ProfilePreview/index.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      fullname: 'Test Fullname',
+      whatIDo: 'coding',
+      status: 'online'
+    },
+    id: '1',
+    onSend: (message: string) => console.log(message),
+    redirectTo: '/profile'
+  };
+  const trigger = () => <Button variant="success">Show</Button>;
+  const [text, setText] = useState('');
```
Comment (vlvereta) [inline frontend/src/components/ProfilePreview/index.tsx:33]: useState should be on the top

File: frontend/src/components/ProfilePreview/index.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  const onSendMessage = (message: string) => {
+    testData.onSend(message);
+    setText('');
+  };
+  function keycheck(e: React.KeyboardEvent<HTMLInputElement>) {
```
Comment (vlvereta) [inline frontend/src/components/ProfilePreview/index.tsx:38]: Create hook for keypress action

File: frontend/src/components/ProfilePreview/index.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import { Link, Route, Redirect } from 'react-router-dom';
+import { faLocationArrow } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { userInfo } from 'os';
+
+interface IProps {
+  user: IUserState;
+  trigger: () => React.ReactElement;
+  id: string;
+  onSend: (message: string) => void;
```
Comment (vlvereta) [inline frontend/src/components/ProfilePreview/index.tsx:15]: Use general Binding callback definition

File: frontend/src/common/models/user/IUser.ts [hunk 5]
```diff
@@ -6,4 +6,5 @@ export interface IUser {
   imageUrl?: string;
   title?: string;
   status?: string;
+  whatIDo?: string;
```
Comment (AlikRakhmonov) [inline frontend/src/common/models/user/IUser.ts:9]: redundant, we have `title` already

File: frontend/src/components/ProfilePreview/index.tsx [hunk 6]
```diff
@@ -17,49 +22,67 @@ interface IProps {
 // Mocked data: testData and trigger function
 const testData = {
```
Comment (AlikRakhmonov) [inline frontend/src/components/ProfilePreview/index.tsx:22]: remove mocks

File: frontend/src/components/ProfilePreview/index.tsx [hunk 7]
```diff
@@ -17,49 +21,60 @@ interface IProps {
...
+// const ProfilePreview: FunctionComponent<IProps> = ({ user, trigger, onSend }) => {
 const ProfilePreview: FunctionComponent = () => {
   const [text, setText] = useState('');
+  const onViewProfile = () => {
+    setUserDataHandler(testData.user);
+    setShowProfileHandler();
+  };
   const onSendMessage = (message: string) => {
     if (text.trim()) {
       testData.onSend(message);
```
Comment (AlikRakhmonov) [inline frontend/src/components/ProfilePreview/index.tsx:47]: use data from props

File: frontend/src/containers/ThreadsContainer/index.tsx [hunk 8]
```diff
@@ -6,18 +6,17 @@ import { IThreadsState } from './reducers/reducer';
...
 export type ISetThreads<T> = (arg: T) => void;
 interface IProps {
-  // fetchThreads: IFetchThreads;
+  fetchThreads: () => void;
   loading: boolean;
 }
 
-const ThreadsContainer: FunctionComponent<IProps> = ({ loading }) => {
+const ThreadsContainer: FunctionComponent<IProps> = ({ fetchThreads, loading }) => {
   const [threads, setThreads]:[IThread[], ISetThreads<IThread[]>] = useState<IThread[]>([]);
```
Comment (vlvereta) [inline frontend/src/containers/ThreadsContainer/index.tsx:16]: I'd rather don't overtype this general hook. You have to write only **useState<IThread[]>([])** and that's all.

File: frontend/src/containers/ThreadsContainer/index.tsx [hunk 9]
```diff
@@ -6,18 +6,17 @@ import { IThreadsState } from './reducers/reducer';
 import Thread from '../Thread';
 import { IThread } from 'common/models/thread/IThread';
 
-export type IFetchThreads = () => IThread[];
 export type ISetThreads<T> = (arg: T) => void;
```
Comment (vlvereta) [inline frontend/src/containers/ThreadsContainer/index.tsx:9]: What the difference from **IBindingCallback1**?

File: frontend/src/components/ProfilePreview/styles.module.sass [hunk 10]
```diff
@@ -38,17 +56,17 @@
   @include status
   background-color: var(--offline)
 
-.whatIDo
+.title
   color: #ffffff
```
Comment (vlvereta) [inline frontend/src/components/ProfilePreview/styles.module.sass:60]: Color.

File: frontend/src/components/ProfilePreview/index.tsx [hunk 11]
```diff
@@ -1,65 +1,60 @@
...
+  // };
   const onSendMessage = (message: string) => {
     if (text.trim()) {
-      testData.onSend(message);
+      onSend(message);
     }
     setText('');
   };
-  function keycheck(e: React.KeyboardEvent<HTMLInputElement>) {
+  const keycheck = (e: React.KeyboardEvent<HTMLInputElement>) => {
```
Comment (vlvereta) [inline frontend/src/components/ProfilePreview/index.tsx:31]: We have a special hook for this, search in the project.

**Pull request merged**